base_bank_from_iban/models/res_partner_bank.py

```diff
@@ -9,10 +13,51 @@
     pretty_iban,
 )
 
+_logger = logging.getLogger(__name__)
+
 
 class ResPartnerBank(models.Model):
     _inherit = "res.partner.bank"
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        vals_list = [self._add_bank_vals(vals) for vals in vals_list]
+        return super().create(vals_list)
+
+    def write(self, vals):
+        self._add_bank_vals(vals)
+        return super().write(vals)
+
+    def _add_bank_vals(self, vals):
+        if vals.get("acc_number") and not vals.get("bank_id"):
+            try:
+                bank = self._get_bank_from_iban(vals["acc_number"])
+                vals["bank_id"] = bank.id
+            except Exception:
+                _logger.info("Could not find bank from IBAN", exc_info=True)
```

Comment:
how do you see controlling a savepoint this part? to prevent exceptions + closed cursors.